x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -445,6 +445,49 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
     };
   }
 
+  /** Purpose of this function is to take a package policy and compile the inputs
+   This is primarily used by the Synthetics UI to display the inputs which are passed to agent
+   Purpose is to debug the inputs which are passed to the agent and also compared them to the config
+   which is passed to public service locations */
+  public async inspect(
+    soClient: SavedObjectsClientContract,
+    packagePolicy: NewPackagePolicyWithId
+  ): Promise<NewPackagePolicy> {
+    if (!packagePolicy.id) {
+      packagePolicy.id = SavedObjectsUtils.generateId();
+    }
+
+    const packageInfos = await getPackageInfoForPackagePolicies([packagePolicy], soClient);
+    const packagePolicyId = packagePolicy.id ?? SavedObjectsUtils.generateId();
```

Comment:
Why call `SavedObjectsUtils.generateId()` two times? Can't we just use `packagePolicy.id` here?